be/src/storage/olap_server.cpp

```diff
@@ -98,19 +98,45 @@ Status StorageEngine::start_bg_threads() {
         vectorized::Compaction::init(max_compaction_concurrency);
 
         _base_compaction_threads.reserve(base_compaction_num_threads);
-        for (uint32_t i = 0; i < base_compaction_num_threads; ++i) {
-            _base_compaction_threads.emplace_back([this, data_dir_num, data_dirs, i] {
-                _base_compaction_thread_callback(nullptr, data_dirs[i % data_dir_num]);
-            });
-            Thread::set_thread_name(_base_compaction_threads.back(), "base_compact");
+        // The config::tablet_map_shard_size is preferably a multiple of `base_compaction_num_threads_per_disk`,
+        // otherwise the compaction thread will be distributed unevenly.
+        int32_t base_step = config::tablet_map_shard_size / base_compaction_num_threads_per_disk +
```

Comment:
```suggestion
        int32_t base_step = (config::tablet_map_shard_size + 1) / base_compaction_num_threads_per_disk;
```

Reply:
if config::tablet_map_shard_size=1, base_comaction_num_thread_per_disk=1, the result should be 1, not 2